language-support/ts/daml-ledger/index.ts

```diff
@@ -953,6 +970,51 @@ class Ledger {
     return decode(partyInfoDecoder, json);
   }
 
+  /**
+   * Fetch a list of all package IDs from the ledger.
+   *
+   * @returns List of package IDs.
+   *
+   */
+  async listPackages(): Promise<PackageId[]> {
+    const json = await this.submit('v1/packages', undefined, 'get');
+    return decode(jtv.array(jtv.string()), json);
+  }
+
+  /**
+   * Fetch a binary package.
+   *
+   * @returns The content of the package as a raw ArrayBuffer.
+   *
+   */
+  async getPackage(id: PackageId): Promise<ArrayBuffer> {
+    const httpResponse = await fetch(this.httpBaseUrl + 'v1/packages/' + id, {
+      headers: this.auth(),
+      method: 'get',
+    });
+    await this.throwOnError(httpResponse);
+    return await httpResponse.arrayBuffer();
+  }
+
+  /**
+   * Upload a binary archive. Note that this requires admin privileges.
+   *
+   * @returns True on success; throws on error.
+   *
+   */
+  async uploadDarFile(abuf: ArrayBuffer): Promise<true> {
+    const httpResponse = await fetch(this.httpBaseUrl + 'v1/packages', {
+      body: abuf,
+      headers: {
+        ...this.auth(),
+        'Content-type': 'application/octet-stream'
+      },
+      method: 'post',
+    });
+    await this.throwOnError(httpResponse);
+    return true;
```

Comment:
```suggestion
    return {};
```
In the APIs I’ve seen, this seemed more common as the unit type but my TS experience is very limited so feel free to disagree with me and keep `true`.